internal/provider/utils/utils.go

```diff
@@ -12,3 +16,16 @@ func NamespacedName(obj client.Object) types.NamespacedName {
 		Name:      obj.GetName(),
 	}
 }
+
+// Returns a partially hashed name for the string if it is more than 60 chars. Otherwise returns the original string
+func GetHashedName(name string) string {
+
+	h := sha256.New() // Using sha256 instead of sha1 due to Blocklisted import crypto/sha1: weak cryptographic primitive (gosec)
+	hsha := h.Sum([]byte(name))
+	hashedName := strings.ToLower(fmt.Sprintf("%x", hsha))
+
+	if len(name) > 32 {
+		return fmt.Sprintf("%s-%s", name[0:32], hashedName[0:8])
+	}
```

Comment:
Would it be more efficient if the func first checks the provided name is less or equal to 32 chars and then return early with the provided name?

Reply:
This bit has been updated to use up to 48 characters so that with the `"envoy-` and the 8 character hash we can go all the way up to 63 characters max. Hashing still occurs on all resources regardless of initial name length at @arkodg's  suggestion but I'm good with doing it either way.